### DISCUSSION TITLE
HasOne::ofMany creating a model

### DISCUSSION THREAD

**Discussion by meyer59** [laravel/nova-issues#4356] [Ideas & Feature Requests] [answered]
- Laravel Version: 9.15
- Nova Version: 4.7
- PHP Version: 8.1
- Database Driver & Version: mysql 8
- Operating System and Version: win10
- Browser type and version: chrome 102 

### Description:
Hello
I am using the laravel `->hasOne(PlayerAmbianceParameters::class)->latestOfMany()` relationship. In nova i am using `HasOne::ofMany` type of field 

So, how i can create this resource in nova? The create button doesnt show up
![image](https://user-images.githubusercontent.com/9505463/171382681-006ca8db-9ed9-48b7-9084-564bf35696b4.png)

**Comment by ianrobertsFF** [comment]
I asked this in the Discord channel, because the documentation is frankly misleadingly suggestive on this point. It's not currently possible to do this. Although they are working on being able to inline create hasMany (which is essentially what a hasOne::ofMany is) with no guarantees it actually ends up in the product any time soon.
 
Documentation currently states basically:

1) HasOne can do Of Many
2) HasOne can create resources inline

At no point does it say: "HasOne can't do both of these things at the same time"

The way I got around it, which is a bit of a code smell, is on my Model to do this:

```php
public function versions() {
        return $this->hasMany(ConfigVersion::class, "config_id");
    }

    public function version_insert() {
        return $this->hasOne(ConfigVersion::class, "config_id");
    }
```

And on my resource to do this:

```php
Fields\HasOne::make('Version', 'version_insert', ConfigVersion::class)->onlyonForms(true)->hideWhenUpdating(true),
Fields\HasMany::make(__("Versions"), "versions", ConfigVersion::class),
```

**Comment by crynobone** [comment, marked as answer]
Has One of Many is essentially a snapshot of Has Many state (eg: Latest Post, Oldest Post, Latest Post from Last Quarter). Therefore it can be considered a read-only field. Furthermore, while we can add a "Create Resource" button and redirect it to the relevant resource create URL, the created resource may not comply with the relationship requirement, especially for Advanced Has One of Many and we still going to have people arguing over this (for bad UI/UX and misleading).

![image](https://user-images.githubusercontent.com/172966/171513683-c66e71e4-e100-400f-8d43-ab42624f5155.png)

**Comment by meyer59** [comment]
i understand your point of view, but blocking users to create the resource in not the best solution. 
Also, for me Has One of Many is more like an HasOne but with history. We want to keed history/old/futur record of the relationship and retrieve only the last interesting one. 
Imo the create button should create the resource as a regular one. Then, when retrieving the record, the relationship will applies whatever flters on it. So if users complain they should alter their relationship `ofmany()` conditions.
Why not letting the user chose to enable or not the creation like ` HasOne::ofMany(...)->enabledCreation(true),` ?
